src/test/TestUtils.h

```diff
@@ -30,10 +30,9 @@ template <typename T = ApplicationImpl>
 std::shared_ptr<T>
 createTestApplication(VirtualClock& clock, Config const& cfg)
 {
-    auto app = Application::create<T>(clock, cfg);
-    auto& lm = app->getLedgerManager();
-    lm.getCurrentLedgerHeader().baseFee = cfg.DESIRED_BASE_FEE;
-    testutil::setCurrentLedgerVersion(lm, cfg.LEDGER_PROTOCOL_VERSION);
+    Config c2(cfg);
+    c2.USE_CONFIG_FOR_GENESIS = true;
```

Comment:
Do we actually want to do it this way? Another option that I see is to update `getTestConfig` in test/test.cpp to have `USE_CONFIG_FOR_GENESIS = true` in all test configurations. The advantage to this won't be realized until #1392 is merged, as that PR changes the type of `Application` that is created by `createTestApplication`. It would therefore be ideal if all tests (possibly excluding simulation tests) could go through `createTestApplication`, which will be impossible for the `LedgerHeader` tests the way you implemented it. 

There is still an issue that `createTestApplication` doesn't support `newDb = false` since I didn't think it made sense to create an application with an existing database but then immediately change the ledger version. This means that there are still some places in the tests where we call `Application::create` instead of `createTestApplication` because we need to use an existing database. Your change will eliminate this issue, so I will fix this `newDb` issue in #1392 once this PR is merged.

Reply:
I thought about that too, I didn't want to introduce a dependency between test configuration (that works in all scenarios) vs what a test application is as we're not creating test applications all the time (simulation is just one of those places like you pointed out).

#1392 should be merged in days, so I am not sure making additional changes to the code to support the interim state really makes sense.
Trying to switch all test cases to use test application is not in scope for this PR that was only looking at fixing the behavior of that factory function.

Reply:
Alright, I am comfortable to accept this solution in the interim since it does resolve the issue that this PR is addressing. But it might become apparent when I do finish going through the process of changing to `createTestApplication` everywhere that we need to do something different. We can handle that when we come to it though. Given that we will do it this way for now, it would be good to address the comment directly above though.